recipes/libusb/all/conanfile.py

```diff
@@ -53,100 +52,136 @@ def config_options(self):
-        return self._autotools
+                    tc.configure_args.append("--build=i686-w64-mingw32")
+                    tc.configure_args.append("--host=i686-w64-mingw32")
+            tc.generate()
+            tc = AutotoolsDeps(self)
+            tc.generate()
+            tc = PkgConfigDeps(self)
+            tc.generate()
+            ms = VirtualBuildEnv(self)
+            ms.generate(scope="build")
+
+    @property
+    def _msvc_sln_filename(self):
+        # INFO: Assume we're using the latest Visual Studio and default to libusb_2019.sln
+        # (or libusb_2017.sln for libusb < 1.0.24).
+        # If we're not using the latest Visual Studio, select an appropriate solution file.
+        solution_msvc_year = 2019 if Version(self.version) >= "1.0.24" else 2017
+        solution_msvc_year = {
+            "11": 2012,
+            "12": 2013,
+            "14": 2015,
+            "15": 2017,
+        }.get(str(self.settings.compiler.version), solution_msvc_year)
+        return os.path.join("msvc", "libusb_{}.sln".format(solution_msvc_year))
+
+    @property
+    def _msvc_build_type(self):
+        return "Debug" if self.settings.build_type == "Debug" else "Release"
+
+    def _msvc_platform(self, msbuild):
+        return "Win32" if self.settings.arch == "x86" else msbuild.platform
+
+    def _patch_msvc_source(self):
+        if Version(self.version) < "1.0.24":
+            for vcxproj in ["fxload_2017", "getopt_2017", "hotplugtest_2017", "libusb_dll_2017",
+                            "libusb_static_2017", "listdevs_2017", "stress_2017", "testlibusb_2017", "xusb_2017"]:
+                vcxproj_path = os.path.join(self.build_folder, "msvc", "%s.vcxproj" % vcxproj)
+                replace_in_file(vcxproj_path, "<WindowsTargetPlatformVersion>10.0.16299.0</WindowsTargetPlatformVersion>", "")
 
     def build(self):
-        if self._is_msvc:
-            if tools.Version(self.version) < "1.0.24":
-                for vcxproj in ["fxload_2017", "getopt_2017", "hotplugtest_2017", "libusb_dll_2017",
-                                "libusb_static_2017", "listdevs_2017", "stress_2017", "testlibusb_2017", "xusb_2017"]:
-                    vcxproj_path = os.path.join(self._source_subfolder, "msvc", "%s.vcxproj" % vcxproj)
-                    tools.replace_in_file(vcxproj_path, "<WindowsTargetPlatformVersion>10.0.16299.0</WindowsTargetPlatformVersion>", "")
-            self._build_visual_studio()
+        if is_msvc(self):
+            self._patch_msvc_source()
+            msbuild = MSBuild(self)
+            msbuild.build_type = self._msvc_build_type
+            msbuild.platform = self._msvc_platform(msbuild)
+            msbuild.build(self._msvc_sln_filename)
         else:
-            autotools = self._configure_autotools()
+            autotools = Autotools(self)
+            try:
+                autotools.autoreconf()
+            except ConanException as error:
+                if "command not found" in str(error):
```

Comment:
There was no call to autoreconf, why is it added?

Reply:
Yes, it's not needed. I'll revert, thanks.